src/pages/settings/Payments/PaymentMethodList.js

```diff
@@ -54,87 +63,75 @@ const defaultProps = {
-                type: MENU_ITEM,
-                title: bankAccount.addressName,
-
-                // eslint-disable-next-line
-                description: formattedBankAccountNumber,
-                icon,
-                iconSize,
-                onPress: e => this.props.onPress(e, bankAccount.bankAccountID),
-                key: `bankAccount-${bankAccount.bankAccountID}`,
-            });
-        });
+            switch (method.type) {
+                case CONST.PAYMENT_METHODS.BANK_ACCOUNT: iconProperties = getBankIcon(method.bankName); break;
+                case CONST.PAYMENT_METHODS.DEBIT_CARD: iconProperties = getBankIcon(method.bankName, true); break;
+                case CONST.PAYMENT_METHODS.PAYPAL: iconProperties = {icon: Expensicons.PayPal}; break;
+                default: break;
+            }
 
-        _.each(this.props.cardList, (card) => {
-            const formattedCardNumber = card.cardNumber
-                ? `${this.props.translate('paymentMethodList.cardLastFour')} ${card.cardNumber.slice(-4)}`
-                : null;
-            const {icon, iconSize} = getBankIcon(card.bank, true);
-            combinedPaymentMethods.push({
+            return {
+                ...method,
+                icon: iconProperties.icon,
+                iconSize: iconProperties.iconSize,
                 type: MENU_ITEM,
-                title: card.addressName,
-                // eslint-disable-next-line
-                description: formattedCardNumber,
-                icon,
-                iconSize,
-                onPress: e => this.props.onPress(e, card.cardNumber),
-                key: `card-${card.cardNumber}`,
-            });
+                onPress: e => this.props.onPress(e, method.id),
+            };
         });
 
-        if (this.props.payPalMeUsername) {
-            combinedPaymentMethods.push({
-                type: MENU_ITEM,
-                title: 'PayPal.me',
-                description: this.props.payPalMeUsername,
-                icon: Expensicons.PayPal,
-                onPress: e => this.props.onPress(e, 'payPalMe'),
-                key: 'payPalMePaymentMethod',
-            });
-        }
-
         // If we have not added any payment methods, show a default empty state
-        if (_.isEmpty(combinedPaymentMethods)) {
+        if (_.isEmpty(paymentMethods)) {
             combinedPaymentMethods.push({
                 text: this.props.translate('paymentMethodList.addFirstPaymentMethod'),
             });
         }
 
+        let addPaymentMethodButtonTitle = this.props.translate('paymentMethodList.addPaymentMethod');
+        switch (this.props.filterType) {
+            case CONST.PAYMENT_METHODS.BANK_ACCOUNT: addPaymentMethodButtonTitle = this.props.translate('paymentMethodList.addBankAccount'); break;
+            case CONST.PAYMENT_METHODS.DEBIT_CARD: addPaymentMethodButtonTitle = this.props.translate('paymentMethodList.addDebitCard'); break;
+            default: break;
+        }
```

Comment:
We should be using the `AddPaymentMethodMenu` for this.

Reply:
But on this page `ChooseTransferAccountPage.js`, only one action is shown based on the type of paymentMethods shown. 
For example, 
In the list is bank accounts, 
then `+ Add Bank Account ` and AddPaymentMethodMenu is a menu with all options.

Reply:
Ok, well we have a component that basically does the same thing, but without being able to change the title. Maybe you can think of some ways to re-use it and the make it do whatever we need with props?